src/pytest_codspeed/plugin.py

```diff
@@ -170,88 +172,58 @@ def pytest_collection_modifyitems(
-    if is_gc_enabled:
-        gc.enable()
-
-    return result
-
-
-@pytest.hookimpl(tryfirst=True)
-def pytest_runtest_protocol(item: pytest.Item, nextitem: pytest.Item | None):
-    plugin = get_plugin(item.config)
-    if not plugin.is_codspeed_enabled or not should_benchmark_item(item):
-        return (
-            None  # Defer to the default test protocol since no benchmarking is needed
-        )
-
-    if has_benchmark_fixture(item):
-        return None  # Instrumentation is handled by the fixture
-
-    plugin.benchmark_count += 1
-    if not plugin.should_measure:
-        return None  # Benchmark counted but will be run in the default protocol
-
-    # Setup phase
-    reports = []
-    ihook = item.ihook
-    ihook.pytest_runtest_logstart(nodeid=item.nodeid, location=item.location)
-    setup_call = pytest.CallInfo.from_call(
-        lambda: ihook.pytest_runtest_setup(item=item, nextitem=nextitem), "setup"
-    )
-    setup_report = ihook.pytest_runtest_makereport(item=item, call=setup_call)
-    ihook.pytest_runtest_logreport(report=setup_report)
-    reports.append(setup_report)
-    # Run phase
-    if setup_report.passed and not item.config.getoption("setuponly"):
-        assert plugin.lib is not None
-        runtest_call = pytest.CallInfo.from_call(
-            lambda: _run_with_instrumentation(
-                plugin.lib, item.nodeid, item.config, item.runtest
-            ),
-            "call",
-        )
-        runtest_report = ihook.pytest_runtest_makereport(item=item, call=runtest_call)
-        ihook.pytest_runtest_logreport(report=runtest_report)
-        reports.append(runtest_report)
-
-    # Teardown phase
-    teardown_call = pytest.CallInfo.from_call(
-        lambda: ihook.pytest_runtest_teardown(item=item, nextitem=nextitem), "teardown"
-    )
-    teardown_report = ihook.pytest_runtest_makereport(item=item, call=teardown_call)
-    ihook.pytest_runtest_logreport(report=teardown_report)
-    reports.append(teardown_report)
-    ihook.pytest_runtest_logfinish(nodeid=item.nodeid, location=item.location)
-
-    return reports  # Deny further protocol hooks execution
+    testfunction: Callable[P, T],
+) -> Callable[P, T]:
+    @functools.wraps(testfunction)
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> T:
+        def __codspeed_root_frame__():
+            return testfunction(*args, **kwargs)
+
+        is_gc_enabled = gc.isenabled()
+        if is_gc_enabled:
+            gc.collect()
+            gc.disable()
+        try:
+            if SUPPORTS_PERF_TRAMPOLINE:
+                # Warmup CPython performance map cache
+                __codspeed_root_frame__()
+
+            lib.zero_stats()
+            lib.start_instrumentation()
+            try:
+                return __codspeed_root_frame__()
+            finally:
+                lib.stop_instrumentation()
+                uri = get_git_relative_uri(nodeid, config.rootpath)
+                lib.dump_stats_at(uri.encode("ascii"))
```

Comment:
since you introduced this try block, can you add a test bench raising an exception?